src/components/core/Input/Input.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { ChangeEvent } from 'react';
+import classes from './Input.module.scss';
+
+interface InputProps {
+  label: string;
+  value: string;
+  type: string;
+  onChange: (value: string) => void;
+}
+
+const Input: React.FC<InputProps> = ({ label, value, type, onChange }) => {
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange(event.target.value);
+  };
+
+  return (
+    <div className={`${classes['c-input-container']}`}>
+      <label className={`${classes['c-input-container__label']}`}>{label}</label>
+      <input
+        className={`${classes['c-input-container__input']}`}
+        type={type}
+        value={value}
+        onChange={handleChange}
```

Comment:
If you receive onChange that use it no need to create wrapper function. Give onChange type of `ChangeEvent<HTMLInputElement>`